zerver/views/invite.py

```diff
@@ -29,13 +29,28 @@
 INVITATION_LINK_VALIDITY_MINUTES: Optional[int] = 24 * 60 * settings.INVITATION_LINK_VALIDITY_DAYS
 
 
-def check_if_owner_required(invited_as: int, user_profile: UserProfile) -> None:
+def check_role_based_permissions(
+    invited_as: int, user_profile: UserProfile, *, require_atleast_admin: bool = False
+) -> None:
     if (
         invited_as == PreregistrationUser.INVITE_AS["REALM_OWNER"]
         and not user_profile.is_realm_owner
     ):
         raise OrganizationOwnerRequiredError
 
+    if require_atleast_admin and not user_profile.is_realm_admin:
+        raise JsonableError(_("Must be an organization administrator"))
+
+    if (
+        invited_as
+        in [
+            PreregistrationUser.INVITE_AS["REALM_ADMIN"],
+            PreregistrationUser.INVITE_AS["MODERATOR"],
+        ]
+        and not user_profile.is_realm_admin
+    ):
+        raise JsonableError(_("Must be an organization administrator"))
```

Comment:
I think this block could be replaced by having the one caller pass `require_admin=invited_as in [PreregistrationUser.INVITE_AS["REALM_ADMIN"], ...]`, maybe with that variable being set on its own line, in the one caller that uses this.